kamailio/listener-defs.cfg

```diff
@@ -10,63 +10,104 @@
-#!substdef "!TCP_EXTERNAL_SIP!tcp:MY_IP_ADDRESS:$def(SIP_PUBLIC_PORT) advertise $def(MY_PUBLIC_IP):$def(SIP_PUBLIC_PORT)!g"
-#!substdef "!TLS_EXTERNAL_SIP!tcp:MY_IP_ADDRESS:$def(SIP_S_PUBLIC_PORT) advertise $def(MY_PUBLIC_IP):$def(SIP_S_PUBLIC_PORT)!g"
-#!substdef "!UDP_EXTERNAL_ALG!udp:MY_IP_ADDRESS:$def(ALG_PUBLIC_PORT) advertise $def(MY_PUBLIC_IP):$def(ALG_PUBLIC_PORT)!g"
-#!substdef "!TCP_EXTERNAL_ALG!tcp:MY_IP_ADDRESS:$def(ALG_PUBLIC_PORT) advertise $def(MY_PUBLIC_IP):$def(ALG_PUBLIC_PORT)!g"
-#!substdef "!TLS_EXTERNAL_ALG!tcp:MY_IP_ADDRESS:$def(ALG_S_PUBLIC_PORT) advertise $def(MY_PUBLIC_IP):$def(ALG_S_PUBLIC_PORT)!g"
-
-#!substdef "!SIP_UDP_LISTENER!UDP_SIP!g"
-#!substdef "!SIP_TCP_LISTENER!TCP_SIP!g"
-#!substdef "!SIP_TLS_LISTENER!TLS_SIP!g"
-#!substdef "!ALG_UDP_LISTENER!UDP_ALG!g"
-#!substdef "!ALG_TCP_LISTENER!TCP_ALG!g"
-#!substdef "!ALG_TLS_LISTENER!TLS_ALG!g"
-
-#!substdef "!WS_TCP_LISTENER!TCP_WS!g"
-#!substdef "!WSS_TLS_LISTENER!TLS_WSS!g"
-
-#!substdef "!SIP_UDP_EXTERNAL_LISTENER!UDP_EXTERNAL_SIP!g"
-#!substdef "!SIP_TCP_EXTERNAL_LISTENER!TCP_EXTERNAL_SIP!g"
-#!substdef "!SIP_TLS_EXTERNAL_LISTENER!TLS_EXTERNAL_SIP!g"
-#!substdef "!ALG_UDP_EXTERNAL_LISTENER!UDP_EXTERNAL_ALG!g"
-#!substdef "!ALG_TCP_EXTERNAL_LISTENER!TCP_EXTERNAL_ALG!g"
-#!substdef "!ALG_TLS_EXTERNAL_LISTENER!TLS_EXTERNAL_ALG!g"
-
-#!substdef "!SIP_UDP_ADVERTISED_LISTENER!UDP_ADVERTISED_SIP!g"
-#!substdef "!SIP_TCP_ADVERTISED_LISTENER!TCP_ADVERTISED_SIP!g"
-#!substdef "!SIP_TLS_ADVERTISED_LISTENER!TLS_ADVERTISED_SIP!g"
-#!substdef "!ALG_UDP_ADVERTISED_LISTENER!UDP_ADVERTISED_ALG!g"
-#!substdef "!ALG_TCP_ADVERTISED_LISTENER!TCP_ADVERTISED_ALG!g"
-#!substdef "!ALG_TLS_ADVERTISED_LISTENER!TLS_ADVERTISED_ALG!g"
-
-#!substdef "!PORT_PROTO_INTERNAL_LISTENER!PROTO_PORT_INTERNAL!g"
+#!substdef "!UDP_EXTERNAL_SIP!udp:MY_IP_ADDRESS:$def(SIP_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(SIP_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!TCP_EXTERNAL_SIP!tcp:MY_IP_ADDRESS:$def(SIP_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(SIP_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!TLS_EXTERNAL_SIP!tcp:MY_IP_ADDRESS:$def(SIP_S_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(SIP_S_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!UDP_EXTERNAL_ALG!udp:MY_IP_ADDRESS:$def(ALG_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(ALG_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!TCP_EXTERNAL_ALG!tcp:MY_IP_ADDRESS:$def(ALG_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(ALG_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!TLS_EXTERNAL_ALG!tcp:MY_IP_ADDRESS:$def(ALG_S_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(ALG_S_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!TCP_EXTERNAL_WS!tcp:MY_IP_ADDRESS:$def(WS_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(WS_EXTERNAL_ADVERTISE_PORT)!g"
+#!substdef "!TLS_EXTERNAL_WSS!tls:MY_IP_ADDRESS:$def(WSS_EXTERNAL_LISTEN_PORT) advertise $def(MY_PUBLIC_IP):$def(WSS_EXTERNAL_ADVERTISE_PORT)!g"
```

Comment:
I now use AWS network load balancer for the only initial requests.
Then Kamailio node inserts its own public IP address as `Record-Route`.
And called sending an ACK to to Kamailio public IP.
From my point of view this more stable.

Also, this approach add the ability to send OPTIONS to check to partners.

Reply:
Also will be good here to use HAproxy protocol for Kamailio.
This will reduce the number of listeners and amount Kamailio process.
But now I faced with the issue https://github.com/kamailio/kamailio/issues/2658

Just for info.